Added Typer.io Presence

dzlandis opened the pull request:
**Name of the service:**
Typer.io

**URL to website:**
https://typer.io

**Screenshots:**
![Pic](https://i.imgur.com/02qesSh.png) ![Pic](https://i.imgur.com/2mxIVzm.png) ![Pic](https://i.imgur.com/GF5zomp.png) ![Pic](https://i.imgur.com/nOkp2dn.png) ![Pic](https://i.imgur.com/7CDhw3i.png) ![Pic](https://i.imgur.com/DzfJktx.png) ![Pic](https://i.imgur.com/2On09ee.png) ![Pic](https://i.imgur.com/AaMEA4x.png) ![Pic](https://i.imgur.com/BhHcbLs.png) ![Pic](https://i.imgur.com/qCoypGs.png) ![Pic](https://i.imgur.com/2mtjBmW.png) ![Pic](https://i.imgur.com/AJXDn2A.png) ![Pic](https://i.imgur.com/LOgZFZG.png) ![Pic](https://i.imgur.com/ruXiCay.png) ![Pic](https://i.imgur.com/a1PWwCa.png)

dzlandis committed:
Added Typer.io Presence

dzlandis commented:
Currently working on fixing Codacy Static Code Analysis issue 👍

dzlandis committed:
Hopefully fixed all Codacy Static Code Analysis issues and rechecked everything works after edits.

dzlandis committed:
Fixed last few Codacy Static Code Analysis problems.

dzlandis committed:
Hopefully fixed last Codacy problem.

dzlandis committed:
Fixed more minor bracket spacing issues with Codacy (hopefully last fix)

dzlandis committed:
Fixed the rest of the bracket stuff, this should be the last.

dzlandis committed:
Curly brace 😳

dzlandis commented:
Codacy Static Code Analysis finally approved 😀

ririxidev commented on websites/T/Typer.io/presence.ts:
```diff
@@ -0,0 +1,180 @@
+    startTimestamp: browsingStamp
+  };
+
+  if (document.location.hostname === "typer.io") {
```
there is no need for this `if` statement.

ririxidev commented on websites/T/Typer.io/dist/metadata.json:
```diff
@@ -0,0 +1,28 @@
+    "version": "1.0.0",
+    "color": "#4e6edd",
+    "tags": [
+      "typing",
+      "multiplayer",
+      "competition",
+      "typerio",
+      "race",
+      "typer",
+      "wpm",
+      "wordsperminute",
+      "words-per-minute"
```
```suggestion
      "typing",
      "competition",
      "typerio",
      "race"
```

rxri reviewed (changes requested):
and run prettier over presence file

dzlandis committed:
Update keywords.

Co-authored-by: ririxidev <mail@ririxi.dev>

dzlandis committed:
Removed unneeded if statement and ran prettier on the presence file.

dzlandis commented:
The DeepScan appears to be failing because of issues on the Apple presence. Nothing I can do about that...

![DeepScan Pic](https://i.imgur.com/MmTr2Sy.png)

ririxidev commented:
I know, just ignore it lol

dzlandis committed:
Extra pretified the code 👍

dzlandis committed:
Fixed comma problems caused by pretify.

dzlandis commented:
Let me know if you'd like me to revert the prettier stuff from this last commit: https://github.com/PreMiD/Presences/pull/3956/commits/f819f455300a180023afb4310bcbd998af629b2b
And revert it back to this one: https://github.com/PreMiD/Presences/pull/3956/commits/c689c99ad9bff6166b23f5c478262d8ce1c6e4f0

https://github.com/PreMiD/Presences/pull/3956/commits/c689c99ad9bff6166b23f5c478262d8ce1c6e4f0 was just fixing indent stuff and https://github.com/PreMiD/Presences/pull/3956/commits/f819f455300a180023afb4310bcbd998af629b2b was actually running the full prettier program.

Let me know which one you prefer. 👍

ririxidev commented:
Well it doesn't look like running prettier because prettier would use tabs for example before `browsingStamp` variable so idk how you did it

dzlandis commented:
> Well it doesn't look like running prettier because prettier would use tabs for example before `browsingStamp` variable so idk how you did it

I think it is possible that I messed things up when copying from https://prettier.io . I'll try it again and fix very soon.

dzlandis committed:
Correctly pretified the file.

dzlandis committed:
Updated queryselector paths for WPM and Accuracy.

I got in contact with the developer of the website and they were kind enough to add some specific hidden HTML paths for me to get WPM and accuracy so it would be accurate 100% of the time. I have asked them to get things like time and race place which I will update once they add it to the website. This is because I realized that the presence wasn't always displaying the most accurate information because of the order in which users join the race. It is very rare that the information be inaccurate but I am still implementing these features so I know it will be accurate 100% of the time.

dzlandis committed:
Ran prettier on the most recent code.

dzlandis commented:
The developer of the website just let me know they added more hidden HTML elements for me to use so the time and race place stuff is accurate 100% of the time. Please don't merge until I make those changes to use them. I'll have  it done by later today. Thanks!

dzlandis committed:
Added more hidden elements and removed no longer needed replace arguments.

dzlandis committed:
Ran prettier on most recent pull request.

dzlandis commented:
That was the last update. I am done and ready to be reviewed. 😀

SlowLife1661 commented on websites/T/Typer.io/presence.ts:
```diff
@@ -0,0 +1,243 @@
+        quickplayWPM !== "undefined" &&
+        quickplayRaceStatus === "Game has Ended"
+      ) {
+        presenceData.details = "The game ended, did not place.";
```
Did not place? This sounds wrong to me

dzlandis replied:
I can change it to "did not finish" if that makes more sense.

SlowLife1661 replied:
yeah well, the context is that they didn't finish the game, is that right

dzlandis replied:
Yep. Fixing.

Slowlife01 reviewed (commented):
the `.split("Jump to")` thing are they necessary?

dzlandis commented:
> the `.split("Jump to")` thing are they necessary?

I'm not 100% but I believe when I tested it, I couldn't get it to work without. I can try again but to my knowledge, it needed that for some reason. I could definitely be wrong though.

dzlandis committed:
Changed "did not place" to "did not finish"

dzlandis committed:
Removed .split("Jump to"[0]

It was no longer needed after my previous updates.

dzlandis committed:
Ran prettier on most recent update.

dzlandis commented:
Should be ready for another check! 👍🏻

SlowLife1661 commented on websites/T/Typer.io/presence.ts:
```diff
@@ -0,0 +1,224 @@
+          "#__next > div.Profile_root__2QIUs > div.Profile_headerContainer__IYvIA > div.Profile_profileContainer__k2Fu9 > div.Profile_wrapper__3Ghk7 > div.Profile_content__rWK4h > h3"
+        )
+        .textContent.trim();
+      presenceData.details = "Viewing a users profile:";
+      presenceData.state = user;
+    } catch {
+      presenceData.details = "Viewing a users profile.";
```
You mean `user`

ririxidev replied:
```suggestion
      presenceData.details = `Viewing a ${user}'s profile:`;
    } catch {
      presenceData.details = "Viewing a user profile.";
```

dzlandis committed:
Updated "viewing a users profile"

Instead of having the username be a separate state, it is now apart of the details area.

Co-authored-by: ririxidev <mail@ririxi.dev>
